code/modules/events/spacevine.dm

```diff
@@ -516,17 +551,17 @@
 /obj/structure/spacevine/proc/spread()
 	var/direction = pick(GLOB.cardinals)
 	var/turf/stepturf = get_step(src,direction)
-	if (!isspaceturf(stepturf) && stepturf.Enter(src))
-		for(var/datum/spacevine_mutation/SM in mutations)
-			SM.on_spread(src, stepturf)
-			stepturf = get_step(src,direction) //in case turf changes, to make sure no runtimes happen
-		if(!locate(/obj/structure/spacevine, stepturf))
+	if(!isspaceturf(stepturf) && stepturf.Enter(src))
+		var/obj/structure/spacevine/spot_taken = locate() in stepturf //Locates any vine on target turf. Calls that vine "spot_taken".
+		var/datum/spacevine_mutation/vine_eating/E = locate() in mutations //Locates the vine eating trait in our own seed and calls it E.
+		if(!spot_taken || (E && (spot_taken && !spot_taken.mutations.Find(E)))) //Proceed if there isn't a vine on the target turf, OR we have vine eater AND the target vine is NOT from our seed.
```

Comment:
Shouldn't you be comparing controller datums here instead of finding same mutation ? Can mutations be shared between controllers ?

Reply:
It's the current vine eater check and I don't fully understand it.  My code comment is wrong so changing it. 

I think it's meant to ensure a vine eater will eat its own tail (from same seed), if the tail itself lacks vine eater. That way it starts eating its old vines once it mutates. Else it will ouroboros itself endlessly. 

What I don't understand is why this check still allows vines from different seeds to eat each other, even if both have vine eater. 

I experimented with a `!(spot_taken.master == master)` check but I can't make it behave better than with the current check. 

Current behavior (live and with this check): 
Vine eater always eats vines from other seeds, but only eats vines from its own seed if it doesn't have vine eater.